.github/workflows/testing.yaml

```diff
@@ -35,79 +48,37 @@ jobs:
         poetry install --remove-untracked
 
     - name: Code formatting (Black)
-      if: always()
+      if: success() || failure()
```

Comment:
Note: Using `success() || failure()` here means that it won't run if the previous job is cancelled, which is usually what we actually want. `always()` kept running after cancellation and lead to some pretty useless results.